### ISSUE TITLE
Server Error passed to error.js without custom properties (appDir)

### ISSUE THREAD

**Issue by pharmarin** [vercel/next.js#48681]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #22 SMP Tue Jan 10 18:39:00 UTC 2023
    Binaries:
      Node: 18.14.2
      npm: 9.5.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.3.1-canary.17
      eslint-config-next: 13.2.4
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/peaceful-leakey-bj0rbx

### To Reproduce

Throw a custom error in server component
Catch error in error.js

### Describe the Bug

The error in client component is missing custom set properties

### Expected Behavior

The error should contain all properties that were set on server side

### Which browser are you using? (if relevant)

Chrome/Safari

### How are you deploying your application? (if relevant)

Vercel

**Comment by DannyPark1** [comment]
This is what im also facing an issue.
if it is unable to know what caused the server side component error on client, it is impossible to handle on specific cases

**Comment by david-alpert-nl** [comment]
this appears to be working as designed, based on the type definition of the return values supported by `getServerSideProps(...)` methods.

- https://github.com/vercel/next.js/discussions/28953#discussioncomment-5689657

I also want more flexibility on handling server-side errors.

**Comment by jpark011** [comment]
According to the [Next.js docs](https://nextjs.org/docs/app/building-your-application/routing/error-handling#securing-sensitive-error-information), properties are removed to prevent any security leaks.

However I agreen with @david-alpert-nl. We need more flexibility when handling errors on RSC.

e.g. I want to show 403 error page (request for login) when an API responses with 403 on RSC.

**Comment by SavioNicodemos** [comment]
Exactly. I'm facing the same issue because I cannot know if is a 404 error or a 403. 
And this way I'm not able to know if a user not exists, or I've just typed the url wrong, or I forgot some param. Should be added more security filters to avoid this kinda of security issue rather then just not show nothing. 
Not even the HTTP status, this is kinda unusual.

**Comment by pbell23** [comment]
Same here. Need to pass custom error informations in RSCs. 
Could we have a custom prop that allows us to pass data that is not considered sentive ?

**Comment by luizrosalba** [comment]
I endorse the request ... we could have a custom field that  explicitly tells user that data is public

**Comment by leonardo2204** [comment]
Totally agree, at least the statusCode would be great

**Comment by luanmm** [comment]
The status code should be sent to the client, otherwise the UX would be really poor. A server error may occur for any HTTP server-side reasons and the client should be able to "translate" the application state to the end-user - something that is very basic to any framework, by the way.

I'm aware about security concerns, but hiding the HTTP status code should not be something that will compromise it in applications. It is important, too, that the actual status code of the response is returned by the server to the client (not sure if it is not yet).

What I think that is a bit "wrong" (by design), having only a guess about the reasons for, is that the error page is required to be a client component, given the fact that the response is something handled by the server. At least it should be a server component, rendered statically in the build process, that receive the HTTP status code and generate specific variations, for instance (that would be client components, if needed to). But I don't know the implementation details or motivations - and I'm not saying that the work here is wrong (please don't think that way by reading this).

Thanks for all the hard work.

**Comment by ntotten** [comment]
This is an extremely unhelpful pattern IMO. There should be some way to access the raw error and choose to show certain properties to the end user. There are many cases where we need to surface error details to the customer in order for them to take some action or at least provide support with more information.

**Comment by sooohka** [comment]
also need option to pass full error object(which is serialzied from server) to error.js,

need this feature to send full error log to third party loggingservice in error.js

**Comment by luanmm** [comment]
@sooohka, this represents security risks, so a better way would be the server itself handling something on its side (if applicable) and sending directly, without the information being ever present on the client-side.

**Comment by marcoshernanz** [comment]
Thanks for the report. This behavior is intentional: Server Component errors cross the React Flight boundary as a new `Error` with a digest, so arbitrary custom properties are not preserved, and production strips server error details to avoid leaks. Expected errors should be handled on the server and returned or rendered as explicit safe data. Broader response status and error signaling work is tracked in #53225, so I’m closing this as not planned.
